mlflow/tracking/client.py

```diff
@@ -1709,6 +1727,54 @@ def log_table(
             # for data like {"inputs": "What is MLflow?"}
             except ValueError:
                 data = pd.DataFrame([data])
+
+        # Check if the column is a `PIL.Image.Image` or `mlflow.Image` object
+        # and save filepath
+        if len(data.select_dtypes(include=["object"]).columns) > 0:
```

Comment:
We should probably move this part out to a helper function for clarity, but can go with a followup PR.